drivers/gpio/gpio_pinctrl_stm32.h

```diff
@@ -1,139 +1,81 @@
-#define STM32_PERIPH_GPIOC LL_APB2_GRP1_PERIPH_GPIOC
-#define STM32_PERIPH_GPIOD LL_APB2_GRP1_PERIPH_GPIOD
-#define STM32_PERIPH_GPIOE LL_APB2_GRP1_PERIPH_GPIOE
-#define STM32_PERIPH_GPIOF LL_APB2_GRP1_PERIPH_GPIOF
-#define STM32_PERIPH_GPIOG LL_APB2_GRP1_PERIPH_GPIOG
-#elif CONFIG_SOC_SERIES_STM32F3X
-#define STM32_CLOCK_BUS_GPIO STM32_CLOCK_BUS_AHB1
-#define STM32_PERIPH_GPIOA LL_AHB1_GRP1_PERIPH_GPIOA
-#define STM32_PERIPH_GPIOB LL_AHB1_GRP1_PERIPH_GPIOB
-#define STM32_PERIPH_GPIOC LL_AHB1_GRP1_PERIPH_GPIOC
-#define STM32_PERIPH_GPIOD LL_AHB1_GRP1_PERIPH_GPIOD
-#define STM32_PERIPH_GPIOE LL_AHB1_GRP1_PERIPH_GPIOE
-#define STM32_PERIPH_GPIOF LL_AHB1_GRP1_PERIPH_GPIOF
-#define STM32_PERIPH_GPIOG LL_AHB1_GRP1_PERIPH_GPIOG
-#define STM32_PERIPH_GPIOH LL_AHB1_GRP1_PERIPH_GPIOH
-#elif CONFIG_SOC_SERIES_STM32F4X
-#define STM32_CLOCK_BUS_GPIO STM32_CLOCK_BUS_AHB1
-#define STM32_PERIPH_GPIOA LL_AHB1_GRP1_PERIPH_GPIOA
-#define STM32_PERIPH_GPIOB LL_AHB1_GRP1_PERIPH_GPIOB
-#define STM32_PERIPH_GPIOC LL_AHB1_GRP1_PERIPH_GPIOC
-#define STM32_PERIPH_GPIOD LL_AHB1_GRP1_PERIPH_GPIOD
-#define STM32_PERIPH_GPIOE LL_AHB1_GRP1_PERIPH_GPIOE
-#define STM32_PERIPH_GPIOF LL_AHB1_GRP1_PERIPH_GPIOF
-#define STM32_PERIPH_GPIOG LL_AHB1_GRP1_PERIPH_GPIOG
-#define STM32_PERIPH_GPIOH LL_AHB1_GRP1_PERIPH_GPIOH
-#define STM32_PERIPH_GPIOI LL_AHB1_GRP1_PERIPH_GPIOI
-#define STM32_PERIPH_GPIOJ LL_AHB1_GRP1_PERIPH_GPIOJ
-#define STM32_PERIPH_GPIOK LL_AHB1_GRP1_PERIPH_GPIOK
-#elif CONFIG_SOC_SERIES_STM32L4X
-#define STM32_CLOCK_BUS_GPIO STM32_CLOCK_BUS_AHB2
-#define STM32_PERIPH_GPIOA LL_AHB2_GRP1_PERIPH_GPIOA
-#define STM32_PERIPH_GPIOB LL_AHB2_GRP1_PERIPH_GPIOB
-#define STM32_PERIPH_GPIOC LL_AHB2_GRP1_PERIPH_GPIOC
-#define STM32_PERIPH_GPIOD LL_AHB2_GRP1_PERIPH_GPIOD
-#define STM32_PERIPH_GPIOE LL_AHB2_GRP1_PERIPH_GPIOE
-#define STM32_PERIPH_GPIOF LL_AHB2_GRP1_PERIPH_GPIOF
-#define STM32_PERIPH_GPIOG LL_AHB2_GRP1_PERIPH_GPIOG
-#define STM32_PERIPH_GPIOH LL_AHB2_GRP1_PERIPH_GPIOH
-#define STM32_PERIPH_GPIOI LL_AHB2_GRP1_PERIPH_GPIOI
-#endif /* CONFIG_SOC_SERIES_.. */
 
-/**
- * @brief configuration of GPIO device
- */
-struct gpio_stm32_config {
-	/* port base address */
-	u32_t *base;
-	/* IO port */
-	int port;
-	struct stm32_pclken pclken;
-};
+#ifndef _GPIO_STM32_H_
+#define _GPIO_STM32_H_
 
-/**
- * @brief driver data
- */
-struct gpio_stm32_data {
-	/* Enabled INT pins generating a cb */
-	u32_t cb_pins;
-	/* user ISR cb */
-	sys_slist_t cb;
-};
+/* -------------------------------------------- */
+/* Driver configuration data from kernel config */
+/* -------------------------------------------- */
+#include <autoconf.h>
 
+/* ------------------------------------------ */
+/* Driver configuration data from device tree */
+/* ------------------------------------------ */
+#include <pinctrl_common.h>
+#include <gpio_common.h>
+#include <dt-bindings/pinctrl/pinctrl_stm32.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <generated_dts_board.h>
```

Comment:
Same as above. Yes I know. It wanted to make it obvious where the driver configuration data is coming from. This is otherwise hidden in the C compiler command line.